Support Rack 3 (attempt #2)

mattbooks opened the pull request:
What is this change doing?
--------
- Support rack v2 and rack v3
- Many updates (ruby, protobuf, envoy, etc.)
- Update capybara and selenium testing (based on sidekiq-gusto and mangled by AI)
- Fix proto generation

Why is this change being made?
--------
- Not many app changes needed to support rack 3 (maybe some of the header lower-casing), however for more modern dependencies lots of updates were needed

How did you test this change?
--------
- CI testing (locally and in github)
- Manual client testing in `gus-gateway`

based on #92 from @ngan and @joemsak

ngan committed:
Support Rack 3

joemsak committed:
Ignore dev env stuff

joemsak committed:
Update Ruby, Node, Yarn, deps, replace apparition with cuprite

joemsak committed:
Remove docker container orphans on rake tasks

joemsak committed:
Upgrade to latest envoy image

joemsak committed:
Update GH actions to use Ruby 3.4

joemsak committed:
Remove frozen bundle line, not working in ruby 3.4

joemsak committed:
Remove --remove-orphans flag

joemsak committed:
Remove 'chromium'; not needed with chrome service

joemsak committed:
Update rubocop syntax

joemsak committed:
Lint

joemsak committed:
WIP trying to upgrade protobuf, grpc, protoc, et al

joemsak committed:
Use customized protoc image to upgrade protoc and the generators

joemsak committed:
Update envoy config

joemsak committed:
Put the apparition code back

joemsak committed:
Rename workflow file, remove rspec

joemsak committed:
Try manual rake cmd

joemsak committed:
Okay, rspec is needed in CI

joemsak committed:
Separate on lines

joemsak committed:
Clean up rakefile

joemsak committed:
Replace apparition with selenium-webdriver, copied from gusto-sidekiq

joemsak committed:
Fix env value type in docker-compose

mattbooks committed:
AI-assisted spec fixes

mattbooks committed:
Fix docker compose version in CI

mattbooks committed:
Ignore cjs/mjs and undo test configs

mattbooks commented:
@SocketSecurity ignore-all

mattbooks committed:
Remove console log printing for feature tests

Removed browser console log printing on test failure.

mattbooks commented on spec/support/capybara.rb:
```diff
@@ -0,0 +1,108 @@
+require 'capybara/rspec'
+require 'selenium-webdriver'
+
+Capybara.register_driver :selenium_chrome_standalone do |app|
```
Some selenium wrangling was required to get the specs working

mattbooks committed:
Remove code for running specs without docker since that does not work

mattbooks committed:
Clean up rakefile

mattbooks committed:
Fix rake constraint and update dependencies for CI

joemsak reviewed (commented):
Looking good so far!

I'm looking into that change in `spec/integration/ruby_server_nodejs_client_spec.rb` in the meantime

mattbooks committed:
Update .rubocop.yml

Co-authored-by: Joe Sak <joe@joesak.com>

joemsak committed:
Fix yarn workspace isolation for node-client (#97)

* Update .gitignore to allow tracking .yarnrc.yml files

Change .yarn* to .yarn/ and .pnp.* to be more specific.
This allows .yarnrc.yml configuration files to be tracked.

* Add .yarnrc.yml files for yarn workspace isolation

Add standalone .yarnrc.yml to spec/node-client/ so Yarn 4 treats it
as an independent project rather than detecting it as a workspace
member of the root project.

* Remove npm install workaround from node client test

With yarn workspace isolation fixed, we no longer need to run
npm install on every test run. The test now just executes the
pre-built node client directly.

- Remove npm install command from test
- Delete package-lock.json (use yarn consistently)
- Update yarn.lock files

* Trigger gh actions
